.gitlab/e2e.yml

```diff
@@ -65,30 +65,48 @@ k8s-e2e-tags-7:
     - DCA_VERSION=$(inv -e cluster-agent.version)
     - inv -e e2e-tests --agent-image=datadog/agent:${AGENT_VERSION} --dca-image=datadog/cluster-agent:${DCA_VERSION}
 
-k8s-e2e-cws:
+.k8s-e2e-cws-cspm-init:
+  - set +x
+  - export DATADOG_AGENT_SITE=datadoghq.com
+  - export DATADOG_AGENT_API_KEY=$(aws ssm get-parameter --region us-east-1 --name ci.datadog-agent.e2e_tests_api_key --with-decryption --query "Parameter.Value" --out text)
+  - export DATADOG_AGENT_APP_KEY=$(aws ssm get-parameter --region us-east-1 --name ci.datadog-agent.e2e_tests_app_key --with-decryption --query "Parameter.Value" --out text)
+
+k8s-e2e-cws-dev:
   extends: .k8s_e2e_template
   rules:
     !reference [.on_dev_branch_manual]
   needs: []
   script:
-    - set +x
-    - export DATADOG_AGENT_SITE=datadoghq.com
-    - export DATADOG_AGENT_API_KEY=$(aws ssm get-parameter --region us-east-1 --name ci.datadog-agent.e2e_tests_api_key --with-decryption --query "Parameter.Value" --out text)
-    - export DATADOG_AGENT_APP_KEY=$(aws ssm get-parameter --region us-east-1 --name ci.datadog-agent.e2e_tests_app_key --with-decryption --query "Parameter.Value" --out text)
+    - !reference [.k8s-e2e-cws-cspm-init]
     - inv -e e2e-tests --agent-image=datadog/agent-dev:${CI_COMMIT_REF_SLUG}-py3 --dca-image=datadog/cluster-agent-dev:${CI_COMMIT_REF_SLUG} --argo-workflow=cws
 
-k8s-e2e-cspm:
+k8s-e2e-cws-main:
+  extends: .k8s_e2e_template
+  rules:
+    !reference [.on_main]
+  needs: []
+  script:
+    - !reference [.k8s-e2e-cws-cspm-init]
+    - inv -e e2e-tests --agent-image=datadog/agent-dev:master-py3 --dca-image=datadog/cluster-agent-dev:master --argo-workflow=cws
+
+k8s-e2e-cspm-dev:
   extends: .k8s_e2e_template
   rules:
     !reference [.on_dev_branch_manual]
   needs: []
   script:
-    - set +x
-    - export DATADOG_AGENT_SITE=datadoghq.com
-    - export DATADOG_AGENT_API_KEY=$(aws ssm get-parameter --region us-east-1 --name ci.datadog-agent.e2e_tests_api_key --with-decryption --query "Parameter.Value" --out text)
-    - export DATADOG_AGENT_APP_KEY=$(aws ssm get-parameter --region us-east-1 --name ci.datadog-agent.e2e_tests_app_key --with-decryption --query "Parameter.Value" --out text)
+    - !reference [.k8s-e2e-cws-cspm-init]
     - inv -e e2e-tests --agent-image=datadog/agent-dev:${CI_COMMIT_REF_SLUG}-py3 --dca-image=datadog/cluster-agent-dev:${CI_COMMIT_REF_SLUG} --argo-workflow=cspm
 
+k8s-e2e-cspm-main:
```

Comment:
Please add your team here https://github.com/DataDog/datadog-agent/blob/31838c53aeac7cd1feeb8809efb0b6f59bbde674/.gitlab/JOBOWNERS#L49-L51 for this job

Reply:
Oh very good point !!